### PR TITLE
Re-add handling of single value headers

### PR DESCRIPTION
**Pull request by nsarychev**
*Issue #, if available:*

*Description of changes:*
This commit https://github.com/awslabs/aws-lambda-go-api-proxy/commit/d7b2b4e73251828d55ff525983683f6e81803866 removed setting single-value headers, which caused all of our tests against gin break. This PR adds back the code to populate single-value headers as it did before. Please consider this for your review. 

Note: There are commits in history from my previous merged PR https://github.com/awslabs/aws-lambda-go-api-proxy/pull/33. I was using same fork just updated it. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: core/response.go [hunk 1]
```diff
@@ -103,8 +103,15 @@ func (r *ProxyResponseWriter) GetProxyResponse() (events.APIGatewayProxyResponse
...
 
+	proxyHeaders := make(map[string]string)
+
+	for h := range r.headers {
+		proxyHeaders[h] = r.headers.Get(h)
+	}
+
 	return events.APIGatewayProxyResponse{
 		StatusCode:        r.status,
+		Headers:         proxyHeaders,
```
Comment (kheppenstall) [inline core/response.go]: Need to run `gofmt` here to line these up?

File: core/response_test.go [hunk 2]
```diff
@@ -154,7 +154,7 @@ var _ = Describe("ResponseWriter tests", func() {
 			Expect(err).To(BeNil())
 
 			// Headers are not written to `Headers` field
```
Comment (kheppenstall) [inline core/response_test.go]: This comment makes it seem like this was intentional. If we want to use `Headers` and `MultiValueHeaders` then this comment does not make sense to keep.

**Comment by nsarychev** [comment]
@sapessi Could you please review this when you get a chance? Thank you!

**Comment by sapessi** [comment]
As soon as the CI for the merge completes I'll tag `v0.6.0`

**Comment by nsarychev** [comment]
Thank you for such a quick response @sapessi!

**Pull request merged**